styleguide/_static/idioms/layering.py

```diff
@@ -0,0 +1,18 @@
+# Good
+from collections import ChainMap
+
+cli_args = {"user": "FooBar", "cmd": "ls -all"}
+env_args = {"user": "Johndoe"}
+cfg_args = {"user": "default", "cwd": "/home/default", "cmd": "ls"}
+
+config = ChainMap(cli_args, env_args, cfg_args)
+
```

Comment:
hm, this needs discussion, the bad example is actually not layering and the ChainMap has actually a quite particular use case of managing hierarchical context